frontend/src/app/projects/models/backup-settings/pipeline-runs/service/pipeline-run.service.ts

```diff
@@ -92,17 +113,58 @@ export class PipelineRunService {
   loadPipelineRuns(
     projectSlug: string,
     modelSlug: string,
-    pipelineID: number
-  ): Observable<PipelineRun[]> {
+    pipelineID: number,
+    page: number,
+    size: number
+  ): Observable<Page<PipelineRun> | undefined | 'loading'> {
+    if (this._pipelineRunPages.getValue().pages[page - 1] !== undefined) {
+      // Skip if already loaded or currently loading
+      return this.getPipelineRunPage(page);
+    }
+
+    this.setPipelineRunPageStatusToLoading(page);
+
     return this.http
-      .get<PipelineRun[]>(this.urlFactory(projectSlug, modelSlug, pipelineID))
+      .get<Page<PipelineRun>>(
+        `${this.urlFactory(
+          projectSlug,
+          modelSlug,
+          pipelineID
+        )}?page=${page}&size=${size}`
+      )
       .pipe(
         tap((pipelineRuns) => {
-          this._pipelineRuns.next(pipelineRuns);
+          const pipelineRunPages = this._pipelineRunPages.getValue();
+          pipelineRunPages.pages[page - 1] = pipelineRuns;
+
+          this.initalizePipelineRunWrapper(pipelineRunPages);
+
+          this._pipelineRunPages.next(pipelineRunPages);
         })
       );
   }
 
+  private initalizePipelineRunWrapper(pageWrapper: PageWrapper<PipelineRun>) {
+    if (pageWrapper.total !== undefined) {
+      // Do nothing, is already initialized
+      return;
+    }
+
+    const firstPage = pageWrapper.pages[0];
+    if (firstPage === undefined || firstPage === 'loading') {
+      throw new TypeError(
+        'first page is undefined, but should be of type Page<PipelineRun>.'
+      );
+    }
+
+    pageWrapper.total = firstPage.total;
+    for (const page of [...Array(pageWrapper.total).keys()]) {
+      if (pageWrapper.pages[page] === undefined) {
+        pageWrapper.pages[page] = undefined;
+      }
+    }
```

Comment:
What is the purpose of this loop, so why do we need it? Because right now this sets `pageWrapper.pages[page]` to undefined only if it is already undefined.